src/client/pydaos/raw/daos_api.py

```diff
@@ -1364,87 +1390,175 @@ def multi_akey_fetch(self, dkey, keys, txn):
-        # Set some default values for
-        # container input parameters.
-        # NOTE: This is not the actual
-        # container properties. These are
-        # input variables which is used
-        # to set appropriate
-        # container properties.
-        super(DaosContProperties, self).__init__()
+        """Initailize a DaosContainerInputParams object.
+
+        Defines the default values for the container input parameters used to
+        set container properties.
+        """
+        super(DaosContainerInputParams, self).__init__()
         self.type = "Unknown"
         self.enable_chksum = False
         self.srv_verify = False
         self.chksum_type = ctypes.c_uint64(100)
         self.chunk_size = ctypes.c_uint64(0)
 
+    def get_daos_properties(self):
+        """Get the DaosProperty defined by this set of container inputs.
+
+        Returns:
+            daos_cref.DaosProperty: the container properties defined by the
+                current input parameters.  Can be None.
+
+        """
+        daos_property = None
+
+        # Define the DaosProperty for the specified inputs:
+        #   1. Layout Type.
+        #   2. Enable checksum,
+        #   3. Server Verfiy
+        #   4. Chunk Size Allocation.
+        if self.type != "Unknown" and not self.enable_chksum:
+            # Only type like posix, hdf5 defined.
+            daos_property = daos_cref.DaosProperty(1)
+        elif self.type == "Unknown" and self.enable_chksum:
+            # Obly checksum enabled.
+            daos_property = daos_cref.DaosProperty(3)
+        elif self.type != "Unknown" and self.enable_chksum:
+            # Both layout and checksum properties defined
+            daos_property = daos_cref.DaosProperty(4)
+
+        if daos_property:
+            # The index is used to increment the dpp_entried array value. If the
+            # layer_type is None and checksum is enabled the index will vary,
+            # e.g.:
+            #   - layer is none, checksum dpp_entries will start with index 0
+            #   - layer is not none, checksum dpp_entries will start at index 1
+            index = 0
+
+            if self.type != "Unknown":
+                # pylint: disable=no-member
+                daos_property.dpp_entries[index].dpe_type = ctypes.c_uint32(
+                    DaosContPropEnum.DAOS_PROP_CO_LAYOUT_TYPE.value)
+
+                if self.type == "posix":
+                    daos_property.dpp_entries[index].dpe_val = ctypes.c_uint64(
+                        DaosContPropEnum.DAOS_PROP_CO_LAYOUT_POSIX.value)
+                elif self.type == "hdf5":
+                    daos_property.dpp_entries[index].dpe_val = ctypes.c_uint64(
+                        DaosContPropEnum.DAOS_PROP_CO_LAYOUT_HDF5.value)
+            else:
+                # pylint: disable=no-member
+                daos_property.dpp_entries[index].dpe_val = ctypes.c_uint64(
+                    DaosContPropEnum.DAOS_PROP_CO_LAYOUT_UNKOWN.value)
```

Comment:
(pylint-no-member) Instance of 'Enum' has no 'DAOS_PROP_CO_LAYOUT_UNKOWN' member